static/js/prepare-form-inputs.js

```diff
@@ -0,0 +1,133 @@
+import intlTelInput from "intl-tel-input";
+
+/**
+ * Initializes phone input field with intlTelInput and pre-fills the country input based on the user's timezone.
+ *
+ * @param {HTMLElement} phoneInput - The input element for the phone number.
+ * @param {HTMLElement} countryInput - The select element for the country.
+ */
+export async function prepareInputFields(phoneInput, countryInput) {
+  const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+  const response = await fetch(`/user-country-tz.json?tz=${timezone}`);
+  if (!response.ok) {
+    throw new Error(`HTTP error with status: ${response.status}`);
+  }
+
+  const data = await response.json();
+  const countryCode = data.country_code || "gb";
+
+  if (phoneInput) {
+    setupIntlTelInput(countryCode, phoneInput);
+  }
+  if (countryInput) {
+    preFormatCountry(countryCode, countryInput);
+  }
+}
+
+/**
+ * Sets the value of the country input field.
+ *
+ * @param {string} countryCode - ISO country code to set as the value.
+ * @param {HTMLElement} countryInput - The select element for the country.
+ */
+function preFormatCountry(countryCode, countryInput) {
+  countryInput.value = countryCode;
+}
+
+
+/**
+ * Configures intlTelInput.
+ *
+ * @param {string} countryCode - ISO country code for initializing the plugin.
+ * @param {HTMLElement} phoneInput - The input element for the phone number.
+ */
+export function setupIntlTelInput(countryCode, phoneInput) {
+  phoneInput.removeAttribute("name"); // Ensure only the hidden input is submitted.
```

Comment:
Just recently fixed this on u.com, removing the input name needs to be done after intlTelInput function. Doing so here and then referencing that name which no longer exists on line 50 will cause the phone field to not be included in the payload on submission for all forms on the site

Reply:
Added the fix, thanks!